reverse_engineering/api.js

```diff
@@ -88,16 +91,17 @@ module.exports = {
 	async getDbCollectionsNames(connectionInfo, logger, callback, app) {
 		try {
 			logInfo('Retrieving databases and tables information', connectionInfo, logger);
+
 			const client = await this.connect(connectionInfo, logger, () => {}, app);
-			if (!client.config.database) {
+			if (!client?.config.database) {
```

Comment:
Is it possible for `await this.connect(...)` to return `null` or `undefined`? Do you need that optional chaining?